harness/determined/cli/trial.py

```diff
@@ -181,7 +181,15 @@ def download(args: Namespace) -> None:
             raise ValueError(f"No checkpoints found for trial {args.trial_id}")
         checkpoint = checkpoints[0]
 
-    path = checkpoint.download(path=args.output_dir)
+    try:
```

Comment:
Printing an error message instead of letting Python render the uncaught exception with a stack trace is helpful. But I think the messaging is still is a little more confusing for the user than it could be

```
% det trial download 3
Invalid Checkpoint State for checkpoint 3ac811f1-d045-4d23-b573-612e853ccbe1. This checkpoint is in state CheckpointState.DELETED. Please retry with a different checkpoint UUID.
```

The person who typed `det trial download` didn't select a UUID, themselves. They _can't_ retry with a different checkpoint UUID because they didn't specify one in the first place.

I think the messaging most helpful here explains what happened during selection and what, if anything, the `det trial download` user can do to fix the problem. If the messaging can explain _why_ things ended up in this state, so much the better. Anything to give the user a sense of control and make it seem less like our system is buggy.

Reply:
I'm not sure I understand, I think all of the information is right there for the user. The checkpoint returned by `--best` is in STATE_DELETED. And to fix it, they can chose a checkpoint with a different UUID.

Reply:
but how would you even pick a different checkpoint UUID? we don't return a list of valid checkpoint UUIDs here, so it's not like we can tell users to call that and get the next best checkpoint UUID. this message might make sense if you specified `--uuid` directly, since that implies you already had some way of finding that UUID, but it's weird for `--best` since i'd assume the CLI would download the best valid checkpoint. 

i think that's what @wes-turner is trying to say. 

i agree that the ideal message will say something about what happened and give reasonable steps to resolve. here i think the best thing the user can do is call `det t ID describe` which will render a list of checkpoints for the trial, and from there, they can manually select the checkpoint they want to download. 

tbh, i would rather just delete this option entirely, and replace it with a `list-checkpoints` on `det trial`, which would just return a list of checkpoints with UUIDs, state, searcher metrics, etc. then the user can call `det c download UUID` for the one they want.

Reply:
> tbh, i would rather just delete this option entirely, and replace it with a list-checkpoints on det trial, which would just return a list of checkpoints with UUIDs, state, searcher metrics, etc. then the user can call det c download UUID for the one they want.

Mmm. I like this strategy. As a long-term one, rather than for this ticket. Which is what I think you meant.